atomicapp/nulecule/main.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+import anymarkup
+import copy
+import logging
+import os
+import shutil
+import uuid
+from string import Template
+
+from atomicapp.constants import (APP_ENT_PATH, EXTERNAL_APP_DIR,
+                                 GLOBAL_CONF, CACHE_DIR)
+from atomicapp.nulecule.lib import NuleculeBase
+from atomicapp.nulecule.container import DockerHandler
+from atomicapp.utils import Utils
+
+logger = logging.getLogger(__name__)
+
+
+class NuleculeManager(object):
+
+    def __init__(self, image, unpack_path=None):
+        self.image = image
+        self.initialize(unpack_path)
+
+    def initialize(self, unpack_path=None):
+        if unpack_path:
+            self.unpack_path = unpack_path
+        else:
+            self.app_name = '{}-{}'.format(
+                Utils.sanitizeName(self.image), uuid.uuid1())
```

Comment:
Seems like following the workflow we have right now..The reason I implemented it like this beforeis that as we used PWD, we need to check the content of PWD before we copy files there to make sure it's the same app (although maybe in a newer version).

May be worth to keep it this way and check the `specversion` before we populate the "real" location of the app